### PR TITLE
Tasers have a brief slowdown period before stun.

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
Tasers now apply slowdown, then stun after a delay.
Antag antistun powers (rejuv, adrenals, epi OD) now removes slowdown aswell.
## Why It's Good For The Game
Tasers are not a healthy mechanic in the game. Many people hate ranged instant stuns because they are far too strong. Clicking on an antag once from a ranged and them losing is not interesting gameplay. This PR helps to reduce this by 1) nerfing how much tasers actually stun from 8-10 seconds to 4-6. and 2) giving the person who is shot by the taser a 4 second window to react with their own stun/getaway option. Sec now actually has to tase then get close and baton them. Hopefully this will reduce the amount of antags that just get game ended by one click. 

This will also help reduce the cliché of a powered vamp getting 3 aegs and being uncatchable by sec because they instantly tase whoever walks near them. 

## Changelog
:cl:
tweak: tasers now stun after a 4 second period of slowdown
tweak: antag antistun powers now remove slowdown
tweak: tasers now have reduced stun times.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/energy.dm [hunk 1]
```diff
@@ -21,11 +19,13 @@
 
 /obj/item/projectile/energy/electrode/on_hit(var/atom/target, var/blocked = 0)
 	. = ..()
+	var/mob/living/carbon/C = target
```
Comment (SteelSlayer) [inline code/modules/projectiles/projectile/energy.dm:22]: Why did you move this up here when it's possible `target` is not carbon?

**Comment by Rurik123** [comment]
This'll be interesting

**Comment by FreeStylaLT** [comment]
img

**Comment by Kepplerx** [comment]
Bubbles you madman. Good luck.

**Comment by Spacemanspark** [comment]
HE'S DOING IT. HE'S GONE MAD, GUYS.

**Comment by MRgaty** [comment]
So vamps and clings and even harder to stop now.
Yeah I dont like this

**Comment by Butlerblock** [comment]
Not that I don’t mind this PR cuz it’d be nice since tasers are dumb, but it definitely need a bigger discussion and third party rebalancing on antags/sec instead of just gutting tasers. Could possibly lead to unforeseen consequences such as much higher win rates on cult(which they def don’t need) or much more chaotic non-team game modes(vamp,tot,cling).

**Comment by hal9000PR** [comment]
> So vamps and clings and even harder to stop now.
> Yeah I dont like this

This is actually untrue. Actually the opposite is true. Tasers are horribly ineffective against clings and vamps, the usual weapon to deal with them is lethal shotguns. Vamps and clings on the otherhand make great use of firing a taser at sec who they are fighting them.

**Comment by MRgaty** [comment]
> > So vamps and clings and even harder to stop now.
> > Yeah I dont like this
> 
> This is actually untrue. Actually the opposite is true. Tasers are horribly ineffective against clings and vamps, the usual weapon to deal with them is lethal shotguns. Vamps and clings on the otherhand make great use of firing a taser at sec who they are fighting them.

I have destroyed vamps and clings with tasers sooooo many times.
A well shot taser so they stay still and three shotgun shots to the face have been my way to go since I started playing sec very very long ago, it works very well.

**Comment by Spacemanspark** [comment]
And that's exactly the type of situation we want to curb.

**Comment by Wojz** [comment]
> A well shot taser so they stay still and three shotgun shots to the face have been my way to go since I started playing sec very very long ago, it works very well.

All you're doing is justifying this PR. 4 seconds is long, I'd go for 2, but this could be tweaked to be shorter if it's too long. Good change, doesn't change our "beloved" stun-based combat THAT much, but makes it more forgiving and less based on latency and sweaty gamer reflexes.

File: code/modules/projectiles/projectile/energy.dm (hunk 1 above)
Comment (hal9000PR) [reply code/modules/projectiles/projectile/energy.dm:22]: uhhhh...... will fix

**Comment by Fox-McCloud** [comment]
As I've detailed elsewhere, the combat system needs to change, but I don't think this is the right direction.

**Comment by hal9000PR** [comment]
Surely this functions as a good way to transition away from tasers until we can completely switch to disablers? Until a PR to switch to disablers is made is this a bad solution?

**Comment by hal9000PR** [comment]
Disabler PR is up so closing in favour of that.
